examples/run/task-env.yaml

```diff
@@ -0,0 +1,61 @@
+---
+apiVersion: pipeline.knative.dev/v1alpha1
+kind: Task
+metadata:
+  name: task-env-merge
+spec:
+  inputs:
+    params:
+    - name: FOO
+      description: FOO variable
+    - name: BAR
+      description: BAR variable
+    - name: FOOBAR
+      description: FOODBAR variable
+      default: foobar
+  steps:
+  # Test the environment variables are set in the build
```

Comment:
there are a few mentions of `build` in the examples that i think would be a bit confusing if you saw the example without understanding the Build -> Task lineage, maybe we can change these to `Task`?